sn_node/src/put_validation.rs

```diff
@@ -254,6 +298,55 @@ impl Node {
         Ok(Some(()))
     }
 
+    async fn register_validation(
+        &self,
+        register: &Register,
+        present_locally: bool,
+    ) -> Result<Option<Register>, ProtocolError> {
+        // check if register is valid
+        let reg_addr = register.address();
+        if let Err(e) = register.verify() {
+            error!("Register with addr {reg_addr:?} is invalid: {e:?}");
+            return Err(ProtocolError::InvalidRegister(*reg_addr));
+        }
+
+        // if we don't have it locally return it
+        if !present_locally {
+            debug!("Register with addr {reg_addr:?} is valid and doesn't exist locally");
+            return Ok(Some(register.to_owned()));
+        }
+        debug!("Register with addr {reg_addr:?} exists locally, comparing with local version");
+
+        // get local register
+        let maybe_record = self
+            .network
+            .get_local_record(&RecordKey::new(reg_addr.name()))
+            .await
+            .map_err(|err| {
+                warn!("Error while fetching local record {err}");
+                ProtocolError::RegisterNotStored(*reg_addr.name())
+            })?;
+        let record = match maybe_record {
+            Some(r) => r,
+            None => {
+                error!("Register with addr {reg_addr:?} already exists locally, but not found in local storage");
+                return Err(ProtocolError::RecordKeyMismatch);
```

Comment:
I think this error was a leftover from a previous change. Maybe we don't need it anymore and we can just return back `RegisterNotStored(*reg_addr.name())`?